src/actions/tree.js

```diff
@@ -35,6 +35,11 @@ export const tree = new Baobab({
   //
   builds: {},
 
+  // the filtered_builds branch of the tree maintains a filtered index of builds nodes
+  // organized by repository owner, name and build number. Only nodes which meet filter
+  // criteria are included.
+  filtered_builds: {},
```

Comment:
could the monkey function go here? I feel like a good amount of the logic can be removed from `events.js` and be placed here with a bit of refactoring and some minor design changes